src/components/icon/index.js

```diff
@@ -0,0 +1,36 @@
+import React, {Component, PropTypes} from 'react';
+
+function renderIcon(name, library, onClick, style) {
+    switch (library) {
+        case 'material':
+            return <i className='material-icons' onClick={onClick} {...style}>{name}</i>;
+        case 'font-awesome':
+            const faCss = `fa fa-${name}`;
+            return <i className={faCss} onClick={onClick} {...style}></i>;
+        case 'font-custom':
+            return <span className={`icon-${name}`}></span>;
+        default:
+            return null;
+    }
+}
+
+function Icon({name, library, onClick, style}) {
+    return(
+        <div>
+            {renderIcon(name, library, onClick, style)}
+        </div>
+    );
+}
+
+Icon.displayName = 'Icon';
+Icon.defaultProps = {
+    name: '',
```

Comment:
that would be better to have default arguments in the function itself rather than with `defaultProps`